libcloud/compute/drivers/linode.py

```diff
@@ -728,6 +752,914 @@ def _to_nodes(self, objs):
+        }
+
+        response = self.connection.request('/v4/images',
+                                           data=json.dumps(attr),
+                                           method='POST').object
+        return self._to_image(response)
+
+    def delete_image(self, image):
+        """Deletes a private image
+
+        :param image: NodeImage to delete (required)
+        :type image: :class:`NodeImage`
+
+        :rtype: ``bool``
+        """
+        if not isinstance(image, NodeImage):
+            raise LinodeExceptionV4("Invalid image instance")
+
+        response = self.connection.request('/v4/images/%s'
+                                           % image.id,
+                                           method='DELETE')
+        return response.status == httplib.OK
+
+    def ex_list_addresses(self):
+        """List IP addresses
+
+        :return: LinodeIPAddress list
+        :rtype: `list` of :class:`LinodeIPAddress`
+        """
+        data = self._paginated_request('/v4/networking/ips', 'data')
+
+        return [self._to_address(obj) for obj in data]
+
+    def ex_list_node_addresses(self, node):
+        """List all IPv4 addresses attached to node
+
+        :param node: Node to list IP addresses
+        :type node: :class:`Node`
+
+        :return: LinodeIPAddress list
+        :rtype: `list` of :class:`LinodeIPAddress`
+        """
+        if not isinstance(node, Node):
+            raise LinodeExceptionV4("Invalid node instance")
+
+        response = self.connection.request('/v4/linode/instances/%s/ips'
+                                           % node.id).object
+        return self._to_addresses(response)
+
+    def ex_allocate_private_address(self, node, address_type='ipv4'):
+        """Allocates a private IPv4 address to node.Only ipv4 is currently supported
+
+        :param node: Node to attach the IP address
+        :type node: :class:`Node`
+
+        :keyword address_type: Type of IP address
+        :type address_type: `bool`
+
+        :return: The newly created LinodeIPAddress
+        :rtype: :class:`LinodeIPAddress`
+        """
+        if not isinstance(node, Node):
+            raise LinodeExceptionV4("Invalid node instance")
+
+        # Only ipv4 is currently supported
+        if address_type != 'ipv4':
+            raise LinodeExceptionV4("Address type not supported")
+        # Only one private IP address can be allocated
+        if len(node.private_ips) >= 1:
+            raise LinodeExceptionV4("Nodes can have up to one private IP")
+
+        attr = {
+            'public': False,
+            'type': address_type
+        }
+
+        response = self.connection.request('/v4/linode/instances/%s/ips'
+                                           % node.id,
+                                           data=json.dumps(attr),
+                                           method='POST').object
+        return self._to_address(response)
+
+    def ex_share_address(self, node, addresses=[]):
```

Comment:
Why setting addresses as an optional keyword?
It is required to share an address.
I think it shoul be better:

 def ex_share_address(self, node, addresses):